cellrank/tl/estimators/mixins/decomposition/_schur.py

```diff
@@ -0,0 +1,300 @@
+
+            For benefits of each method, see :class:`pygpcca.GPCCA`.
+        %(eigen)s
+
+        Returns
+        -------
+        Nothing, just updates the following fields:
+
+            - :attr:`schur_vectors` - %(schur_vectors.summary)s
+            - :attr:`schur_matrix` -  %(schur_matrix.summary)s
+            - :attr:`eigendecomposition` - %(eigen.full_desc)s
+        """
+        if n_components < 2:
+            logg.warning(
+                f"Number of Schur vectors `>=2`, but only `{n_components}` "
+                f"were requested. Using `n_components=2`"
+            )
+            n_components = 2
+
+        if method not in ("brandts", "krylov"):
+            raise ValueError(
+                f"Invalid method `{method!r}`. Valid options are:`'brandts'` or `'krylov'`."
+            )
+
+        try:
+            import petsc4py
+            import slepc4py
+        except ImportError:
+            method = "brandts"
+            logg.warning(
+                f"Unable to import `petsc4py` or `slepc4py`. Using `method={method!r}`"
+            )
+
+        tmat = self.transition_matrix
+        if method == "brandts" and issparse(self.transition_matrix):
+            logg.warning("For `method='brandts'`, dense matrix is required. Densifying")
+            tmat = tmat.A
+
+        self._gpcca = GPCCA(tmat, eta=initial_distribution, z=which, method=method)
+        start = logg.info("Computing Schur decomposition")
+
+        try:
+            self._gpcca._do_schur_helper(n_components)
+        except ValueError:
+            logg.warning(
+                f"Using `{n_components}` components would split a block of complex conjugates. "
+                f"Using `n_components={n_components + 1}`"
+            )
+            self._gpcca._do_schur_helper(n_components + 1)
+
+        self._invalid_n_states = np.array(
+            [
+                i
+                for i in range(2, len(self._gpcca._p_eigenvalues))
+                if _check_conj_split(self._gpcca._p_eigenvalues[:i])
+            ]
+        )
+        if len(self._invalid_n_states):
+            logg.info(
+                f"When computing macrostates, choose a number of states NOT in `{list(self._invalid_n_states)}`"
+            )
+
+        self._write_schur_decomposition(
+            {
+                "D": self._gpcca._p_eigenvalues,
+                "eigengap": _eigengap(self._gpcca._p_eigenvalues, alpha),
+                "params": {
+                    "which": which,
+                    "k": len(self._gpcca._p_eigenvalues),
+                    "alpha": alpha,
+                },
+            },
+            vectors=self._gpcca._p_X,
+            matrix=self._gpcca._p_R,
+            params=self._create_params(),
+            time=start,
+        )
+
+    @d.dedent
+    def plot_schur_matrix(
+        self,
+        title: Optional[str] = "schur matrix",
+        cmap: str = "viridis",
+        figsize: Optional[Tuple[float, float]] = None,
```

Comment:
```suggestion
        figsize: Tuple[Optional[float], Optional[float]] = None,
```

Reply:
Correct type is `Optional[Tuple[float, float]]`, not changed.